packages/babel-plugin-transform-nullish-coalescing-operator/src/index.js

```diff
@@ -0,0 +1,38 @@
+import syntaxNullishCoalescingOperator from "@babel/plugin-syntax-nullish-coalescing-operator";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxNullishCoalescingOperator,
+
+    visitor: {
+      LogicalExpression(path) {
+        const { node, scope } = path;
+        if (node.operator !== "??") {
+          return;
+        }
+
+        const ref = scope.generateUidIdentifierBasedOnNode(node.left);
+        scope.push({ id: ref });
```

Comment:
A possible future optimization is to try to reuse the ref if they are in the same scope and the ref does not escape.

I don't know if the scope API supports escape analysis, though.